python/TestHarness/util.py

```diff
@@ -414,6 +425,66 @@ def getSortedValuesSets(self):
             d = dict(((k, v-t) for k, v in d.items() if v))
         return r
 
+class ReverseReachability:
+    """
+    Calculates the Reverse Reachability of a graph. This class does this by
+    computing the reachability of a graph (reversing the direction of edges)
+    """
+    def __init__(self):
+        """
+        Creats the dictionary to hold all of the reverse dependencies. (i.e. the key -> value
+        pair is inserted in the dictionary such that the value -> key)
+        """
+        self.dependency_dict = {}
+
+    def insertDependency(self, key, values):
+        """
+        Inserts all of the dependencies for a given key. "values" can either be
+        a single value or a set of values.
+
+        """
+        for value in values:
+            default_set = set()
+            default_set.add(key)
+            self.dependency_dict.setdefault(value, default_set).add(key)
```

Comment:
Why not `self.dependency_dict.setdefault(value, set()).add(key)` and get rid of the previous 2 lines?
If the key "value" doesn't exist it just creates an empty set and adds value "key" to it, which is what your previous 2 lines do.